view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -639,9 +646,94 @@ define(
                 }
                 return null;
             }),
-        isIconEnabled: function () {
-            return window.checkoutConfig.payment.adyen.showLogo;
-        },
+            isIconEnabled: function () {
+                return window.checkoutConfig.payment.adyen.showLogo;
+            },
+            /**
+             * Based on the response we can start a action component or redirect
+             * @param responseJSON
+             */
+            validateActionOrPlaceOrder: function (responseJSON, orderId) {
+                var self = this;
+                var response = JSON.parse(responseJSON);
+
+                if (!!response.action) {
+                    // render component
+                    self.orderId = orderId;
+                    self.renderActionComponent(response.action);
+                } else {
+                    $.mage.redirect(
+                        window.checkoutConfig.payment[quote.paymentMethod().method].redirectUrl
+                    );
+                }
+            },
+            /**
+             * Rendering the 3DS2.0 components
+             * To do the device fingerprint at the response of IdentifyShopper render the threeDS2DeviceFingerprint
+             * component
+             * To render the challenge for the customer at the response of ChallengeShopper render the
+             * threeDS2Challenge component
+             * Both of them is going to be rendered in a Magento dialog popup
+             *
+             * @param type
+             * @param token
+             */
+            renderActionComponent: function (action) {
+                var self = this;
+                var actionNode = document.getElementById('ActionContainer');
+
+
+                fullScreenLoader.stopLoader();
+
+                self.popupModal = $('#ActionModal').modal({
+                    // disable user to hide popup
+                    clickableOverlay: false,
+                    responsive: true,
+                    innerScroll: false,
+                    // empty buttons, we don't need that
+                    buttons: [],
+                    modalClass: 'ActionModal'
+                });
+
+                self.popupModal.modal("openModal");
+                self.actionComponent = self.checkoutComponent.createFromAction(action).mount(actionNode);
+            },
+            handleOnAdditionalDetails: function (state, component) {
+
+                var self = this;
+
+                // call endpoint with state.data
+                debugger;
+                var request = state.data;
+                request.orderId = self.orderId;
+
+                threeds2.processThreeDS2(request).done(function () {
```

Comment:
let's add a comment for future reference that now the threeDS2Processor handles the actions as well so we won't forget that this is already here when we refactor this

Reply:
sure can do.